node/src/utils.rs

```diff
@@ -120,6 +125,45 @@ pub(crate) fn leak<T>(value: T) -> &'static T {
     Box::leak(Box::new(value))
 }
 
+/// A flag shared across multiple subsystem.
+#[derive(Copy, Clone, DataSize, Debug)]
+pub(crate) struct SharedFlag(&'static AtomicBool);
+
+impl SharedFlag {
+    /// Creates a new shared flag.
+    ///
+    /// The flag is initially not set.
+    pub(crate) fn new() -> Self {
+        SharedFlag(leak(AtomicBool::new(false)))
+    }
+
+    /// Checks whether the flag is set.
+    pub(crate) fn is_set(self) -> bool {
+        self.0.load(Ordering::SeqCst)
+    }
+
+    /// Set the flag.
+    pub(crate) fn set(self) {
```

Comment:
Should we have `unset` method for completeness?

Reply:
No, I left it out for now, since these flags should be "set once", otherwise we may have to think about scenarios where some clients miss a set/unset cycle.